src/IO/H5/Python/Dat.cpp

```diff
@@ -0,0 +1,56 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include <Python.h>
+#include <boost/python.hpp>
+#include <boost/python/stl_iterator.hpp>
+#include <boost/python/suite/indexing/vector_indexing_suite.hpp>
+#include <hdf5.h>
+#include <sstream>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "DataStructures/Matrix.hpp"
+#include "IO/H5/Dat.hpp"
+#include "IO/H5/Header.hpp"
+#include "IO/H5/OpenGroup.hpp"
+#include "IO/H5/Python/Numpy.hpp"
+#include "IO/H5/Python/ToNumpy.hpp"
+#include "IO/H5/Version.hpp"
+#include "PythonBindings/VectorPyList.hpp"
+
+namespace bp = boost::python;
+
+namespace py_bindings {
+void bind_h5dat() {
+  // Wrapper for basic H5Dat operations
+  bp::class_<h5::Dat, boost::noncopyable>("H5Dat", bp::no_init)
```

Comment:
```cpp
bp::class_<h5::Dat, boost::shared_ptr<h5::Dat>, boost::noncopyable>
```

https://stackoverflow.com/questions/36262320/boost-python-return-a-list-of-noncopyable-objects

Reply:
So I think in general this idea is functional, but in this particular case it won't be enough.  It probably fixes the `__enter__` method, but it turns out that the `__exit__` method  takes 3 arguments (`value`, `type` ,`traceback`) , and without boost documentation (which I can't find anywhere on the `__exit__` method) it seems incredibly unlikely that these can actually be translated back and forth via boost in a way that can actually be understood and predicted.

Reply:
Okay, let's leave this off and someone can revisit it in the future. I think we need to prioritize getting the rest of your work into the code :)